File: library/src/main/java/io/vithor/yamvpframework/mvp/presenter/sketch/Sketch.java
package io.vithor.yamvpframework.mvp.presenter.sketch;

import android.support.annotation.MainThread;

import io.vithor.yamvpframework.mvp.presenter.PresenterAction;

/**
 * Created by Vithorio Polten on 1/8/16.
 */
@MainThread
public interface Sketch {
    /**
     * Display a loading view while loading data in background.
     * <b>The loading view must have the id = R.id.loadingView</b>
     *
     * @param action true, if pull-to-refresh has been invoked loading.
     */
    void showLoading(PresenterAction action);

    /**
     * Show the content view.
     *
     * <b>The content view must have the id = R.id.contentView</b>
     */
    void showContent();

    /**
     * Show the error view when a Network Error occurs.
     *
     * @param pullToRefresh true, if the exception was thrown during pull-to-refresh, otherwise
     * false.
     */
    void showNetworkError(boolean pullToRefresh);

    /**
     * Show the error view when a Server Error occurs.
     *
     * @param pullToRefresh true, if the exception was thrown during pull-to-refresh, otherwise
     * false.
     */
    void showServerError(boolean pullToRefresh);
}

File: mvp/src/main/java/io/vithor/yamvpframework/mvp/presenter/sketch/Sketch.kt
package io.vithor.yamvpframework.mvp.presenter.sketch

import android.support.annotation.MainThread
import io.vithor.yamvpframework.mvp.presenter.Presenter

import io.vithor.yamvpframework.mvp.presenter.PresenterAction

/**
 * Created by Vithorio Polten on 1/8/16.
 */
@MainThread
interface Sketch : Presenter.Contract.View {
    /**
     * Display a loading view while loading data in background.
     * **The loading view must have the id = R.id.loadingView**

     * @param action true, if pull-to-refresh has been invoked loading.
     */
    fun showLoading(action: PresenterAction)

    /**
     * Show the content view.

     * **The content view must have the id = R.id.contentView**
     */
    fun showContent()

    /**
     * Show the error view when a Network Error occurs.

     * @param pullToRefresh true, if the exception was thrown during pull-to-refresh, otherwise
     * * false.
     */
    fun showNetworkError(pullToRefresh: Boolean)

    /**
     * Show the error view when a Server Error occurs.

     * @param pullToRefresh true, if the exception was thrown during pull-to-refresh, otherwise
     * * false.
     */
    fun showServerError(pullToRefresh: Boolean)
}
